File: CodeForces/ECR_63_2_A.py

# -*- coding: utf-8 -*-
# @Date    : 2019-04-25 08:35:28
# @Author  : raj lath (oorja.halt@gmail.com)
# @Link    : link
# @Version : 1.0.0

import sys
sys.setrecursionlimit(10**5+1)

inf     =  int(10 ** 20)
max_val =  inf
min_val = -inf

RW  = lambda : sys.stdin.readline().strip()
RI  = lambda : int(RW())
RMI = lambda : [int(x) for x in sys.stdin.readline().strip().split()]
RWI = lambda : [x for x in sys.stdin.readline().strip().split()]


lens = RI()
ins  = RW()
valid = False
for i in range(lens - 1):
    if ins[i] > ins[i + 1]:
        print("YES")
        print(*[i+1, i + 2])
        valid = True
        break
if not valid : print("NO")

File: CodeForces/ECR63_2_A.kt
fun main(args: Array<String>) {
    solve()
}

private fun read(delimit: Char = ' ') = readLine()!!.split(delimit)

private fun solve() {
    read()
    var s = read()[0]

    var idx = 0
    val length = s.length - 1
    var found = false

    while (idx < length) {
        var s1: String = s[idx] + "" + s[idx + 1]
        if (s1 > s1.reversed()) {
            found = true
            break
        }
        idx++
    }

    if (!found) {
        println("NO")
    } else {
        println("YES")
        println("${idx + 1} ${idx + 2}")
    }
}
